integration/integration_test_utils.py

```diff
@@ -68,40 +73,101 @@ def __init__(self, **kwargs):
+
+def redis_driver(redis_fake_server = None, *args, **kwargs):
+    return RedisDriver(redis_client = get_redis_client(redis_fake_server),key_prefix="storey-test:", *args, **kwargs)
+
+def get_driver(setup_teardown_test, *args, **kwargs):
+    if setup_teardown_test["driver_name"] == "V3ioDriver":
+        return V3ioDriver(*args, **kwargs)
+    elif setup_teardown_test["driver_name"] == "RedisDriver":
+        redis_fake_server = setup_teardown_test["redis_fake_server"] if "redis_fake_server" in setup_teardown_test else None
+        return redis_driver(redis_fake_server = redis_fake_server, *args, **kwargs)
+    else:
+        assert 0
+
+def remove_redis_table(table_name):
+    redis_client = get_redis_client()
+    count = 0
+    ns_keys = "storey-test:" + table_name + "*"
+    for key in redis_client.scan_iter(ns_keys):
+        redis_client.delete(key)
+        count += 1
+
+drivers_list = ["V3ioDriver", "RedisDriver"]
+@pytest.fixture(params=drivers_list)
+def setup_teardown_test(request):
     # Setup
     table_name = _generate_table_name()
+    
+    test_params = {}
+    driver_name = request.param
+    test_params["driver_name"] = driver_name
+    test_params["table_name"] = table_name
+    if driver_name == "RedisDriver":
+        REDIS_URL = os.environ.get('REDIS_URL')
+        if not REDIS_URL:
+            test_params["redis_fake_server"] = fakeredis.FakeServer()
 
     # Test runs
-    yield table_name
+    yield test_params
 
     # Teardown
-    asyncio.run(recursive_delete(table_name, V3ioHeaders()))
+    if driver_name == "V3ioDriver":
+        asyncio.run(recursive_delete(table_name, V3ioHeaders()))
+    elif driver_name == "RedisDriver":
+        remove_redis_table(table_name)
+    else:
+        assert 0
 
 
-@pytest.fixture()
-def setup_kv_teardown_test():
+@pytest.fixture(params=drivers_list)
+def setup_kv_teardown_test(request):
     # Setup
-    table_path = _generate_table_name()
-    asyncio.run(create_temp_kv(table_path))
+    table_name = _generate_table_name()
+    
+    test_params = {}
+    driver_name = request.param
+    test_params["driver_name"] = driver_name
+    test_params["table_name"] = table_name
+    if driver_name == "RedisDriver":
+        REDIS_URL = os.environ.get('REDIS_URL')
+        if not REDIS_URL:
+            test_params["redis_fake_server"] = fakeredis.FakeServer()
+
+    if driver_name == "V3ioDriver":
+        asyncio.run(create_temp_kv(table_name))
+    elif driver_name == "RedisDriver":
+        create_temp_redis_kv(test_params)
+    else:
+        assert 0
 
     # Test runs
-    yield table_path
+    yield test_params
 
     # Teardown
-    asyncio.run(recursive_delete(table_path, V3ioHeaders()))
+    if driver_name == "V3ioDriver":
+        asyncio.run(recursive_delete(table_name, V3ioHeaders()))
+    elif driver_name == "RedisDriver":
+        remove_redis_table(table_name)
+    else:
+        assert 0
```

Comment:
Here too we need to raise an error with what happened as mentioned above. Not just `assert False` (let alone 0...).

Reply:
fixed